torch/csrc/jit/passes/remove_dropout.cpp

```diff
@@ -0,0 +1,54 @@
+#include <torch/csrc/jit/passes/remove_dropout.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+bool isDropoutRemovable(const Node* node) {
+  const auto inputs = node->inputs();
+  TORCH_INTERNAL_ASSERT(inputs.size() == 3);
+  const Value* training_input = inputs[2];
+  auto optional_ivalue = toIValue(training_input);
+  TORCH_INTERNAL_ASSERT(optional_ivalue.has_value());
```

Comment:
This is going to crash the program when `train` isn't a constant, isn't it? e.g.

```
import torch
import torch.nn.functional as F

class TestDropout(torch.nn.Module):
    def forward(self, x, training : bool):
        return torch.dropout(x, 0.5, training)

dropout_training = torch.jit.script(TestDropout())

print(dropout_training.graph)
```

```
graph(%self : __torch__.TestDropout,
      %x.1 : Tensor,
      %training.1 : bool):
  %4 : float = prim::Constant[value=0.5]() # dropout.py:6:32
  %6 : Tensor = aten::dropout(%x.1, %4, %training.1) # dropout.py:6:15
  return (%6)
```

Reply:
Yes, indeed. Thanks for pointing this out. Will put guard around it.

Reply:
This was intended to be deployed post freezing, but still, this has to be self-contained so will fix this.

Reply:
Probably the best course is to not remove such dropout and just leave a warning that it was not removed.